docs/tutorials/constraints.rst

```diff
@@ -0,0 +1,238 @@
+  using the `astroplan.constraints.AirmassConstraint` class.
+
+* Since we're optical observers, we only want to observe targets at night, so
+  we'll also call the `astroplan.constraints.AtNightConstraint` class. We're
+  not terribly worried about sky brightness for these bright stars, so we'll
+  define "night" times as those between civil twilights by using the class
+  method `AtNightConstraint.twilight_civil`::
+
+    from astroplan import (AltitudeConstraint, AirmassConstraint,
+                           AtNightConstraint)
+    constraints = [AltitudeConstraint(10*u.deg, 80*u.deg),
+                   AirmassConstraint(5), AtNightConstraint.twilight_civil()]
+
+This list of constraints can now be applied to our target list to determine
+whether:
+
+* the targets are observable given the constraints at *any* times in the time
+  range, using `astroplan.constraints.is_observable`,
+
+* the targets are observable given the constraints at *all* times in the time
+  range, using `astroplan.constraints.is_always_observable`::
+
+    from astroplan import is_observable, is_always_observable
+    # Are targets *ever* observable in the time range?
+    ever_observable = is_observable(constraints, subaru, targets, time_range=time_range)
+
+    # Are targets *always* observable in the time range?
+    always_observable = is_always_observable(constraints, subaru, targets, time_range=time_range)
+
+These two functions will return boolean arrays which tell you whether or not
+each target is observable given your constraints. Let's print these results in
+tabular form:
+
+    >>> from astropy.table import Table
+    >>> import numpy as np
+    >>> observability_array = np.array([[t.name, ever, always]  for t, ever, always in
+    ...                                 zip(targets, ever_observable, always_observable)])
+    >>> observability_table = Table(observability_array,
+    ...                             names=('Target', 'Ever Observable',
+    ...                                    'Always Observable'))
+    >>> print(observability_table)
+    <Table length=6>
+     Target  Ever Observable Always Observable
+    unicode7     unicode7         unicode7
+    -------- --------------- -----------------
+     Polaris            True              True
+        Vega            True              True
+     Albireo            True             False
+       Algol            True             False
+       Rigel           False             False
+     Regulus           False             False
+
+Now we can see which targets are observable!
+
+.. _constraints-user_defined_constraints:
+
+User-Defined Constraints
+========================
+
+There are many possible constraints that you could find useful which have not
+been implemented (yet) in astroplan. This example will walk you through creating
+your own constraint which will be compatible with the tools in the `constraints`
+module.
```

Comment:
Again, there is no user-visible `constraint` module at the moment, so the docs shouldn't talk about it.

Just in case you're leaning that way: I'm 50:50 to make it a user-visible sub-module, either way is fine with me.